### PR TITLE
Adding index on user.roles to speed up roles-related queries.

### PR DESCRIPTION
**Pull request by aoranwu**
I add a GIN index on roles field of users table to speed up roles-related queries. Is this the right type of index to use for JSONB-typed field?

Confirmed in Neon: image

### CODE REVIEW

File: src/api/database/models/User.ts [hunk 1]
```diff
@@ -37,6 +38,10 @@ class User extends ParanoidModel<
   @Column(STRING)
   clientId: string;
 
+  @Index({
+    name: 'user-roles-index',
```
Comment (weihanwang) [inline src/api/database/models/User.ts:42]: Do you really need to specify the name?

**Comment by weihanwang** [comment]
@aoranwu Did you verify the working of the index by using `sql explain` or alike? In the code we use sequelize's Op.contains to search into the column:

https://github.com/yuanjian-org/app/blob/3c1998396f5975d2a0e9e50c9f3607242298e59b/src/api/sendgrid.ts#L83

https://github.com/yuanjian-org/app/blob/3c1998396f5975d2a0e9e50c9f3607242298e59b/src/api/routes/users.ts#L129

**Comment by aoranwu** [comment]
> @aoranwu Did you verify the working of the index by using `sql explain` or alike? In the code we use sequelize's Op.contains to search into the column:
> 
> https://github.com/yuanjian-org/app/blob/3c1998396f5975d2a0e9e50c9f3607242298e59b/src/api/sendgrid.ts#L83
> 
> https://github.com/yuanjian-org/app/blob/3c1998396f5975d2a0e9e50c9f3607242298e59b/src/api/routes/users.ts#L129

The indexed is used when I inserted ~5000 fake data entries: 
image

When there're fewer records in the database (~1000, for example), the query performs sequential scan without using the index.

File: src/api/database/models/User.ts (hunk 1 above)
Comment (aoranwu) [reply src/api/database/models/User.ts]: 'user.roles' will be auto-generated to be the name of the index so removed.

**Review by weihanwang — APPROVED** [review]

**Pull request merged**